src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -259,4 +259,20 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       " ORDER BY v.sessionSlot.slotStart,v.id",
   )
   fun findBookedVisitsForDate(prisonCode: String, date: LocalDate): List<Visit>
+
+  @Query(
+    "Select v.reference FROM visit v " +
+      " LEFT JOIN event_audit ea on ea.booking_reference = v.reference and ea.type in ('UPDATED_VISIT','BOOKED_VISIT') " +
+      " LEFT JOIN actioned_by ab on ab.id = ea.actioned_by_id" +
+      " LEFT JOIN session_slot ss on ss.id = v.session_slot_id " +
+      " WHERE ab.booker_reference = :bookerReference AND v.visit_status = 'BOOKED' AND ss.slot_date >= CURRENT_DATE AND " +
+      "      v.user_type = 'PUBLIC' AND v.visit_status = 'BOOKED' GROUP BY v.reference",
+    nativeQuery = true,
+  )
+  fun getPublicFutureBookingReferenceByBookerReference(bookerReference: String): List<String>
+
+  @Query(
+    "SELECT v FROM Visit v WHERE v.reference in :bookingReferences order by v.modifyTimestamp",
+  )
+  fun findVisitsByReferences(bookingReferences: List<String>): List<Visit>
```

Comment:
why do we take a list of bookerReferences?

Reply:
typo